frontend/src/lib/components/PropertyFilters/propertyFilterLogic.ts

```diff
@@ -63,56 +57,21 @@ export const propertyFilterLogic = kea<propertyFilterLogicType>({
         update: () => {
             const cleanedFilters = [...values.filters].filter(isValidPropertyFilter)
 
-            // if the last filter is used, add an empty filter to get the "new filter" button
-            if (isValidPropertyFilter(values.filters[values.filters.length - 1])) {
-                actions.newFilter()
-            }
-
-            if (props.onChange) {
-                props.onChange(cleanedFilters)
-            } else {
-                const { [`${props.urlOverride || 'properties'}`]: properties, ...searchParams } =
-                    router.values.searchParams // eslint-disable-line
-                const { pathname } = router.values.location
-
-                searchParams[props.urlOverride || 'properties'] = cleanedFilters
-
-                if (!objectsEqual(properties, cleanedFilters)) {
-                    router.actions.replace(pathname, searchParams)
-                }
-            }
-        },
-    }),
-
-    urlToAction: ({ actions, values, props }) => ({
-        '*': (_, { [`${props.urlOverride || 'properties'}`]: properties }) => {
-            if (props.onChange) {
-                return
-            }
-            let filters
-            try {
-                filters = values.filters
-            } catch (error) {
-                // since this is a catch-all route, this code might run during or after the logic was unmounted
-                // if we have an error accessing the filter value, the logic is gone and we should return
-                return
-            }
-            properties = parseProperties(properties)
-
-            if (!objectsEqual(properties || {}, filters)) {
-                // {} adds an empty row, which shows 'New Filter'
-                actions.setFilters(properties ? [...properties, {}] : [{}])
-            }
+            props.onChange(cleanedFilters)
         },
     }),
 
     selectors: {
         filledFilters: [(s) => [s.filters], (filters) => filters.filter(isValidPropertyFilter)],
+        filtersWithNew: [
+            (s) => [s.filters],
+            (filters) => {
+                if (filters.length === 0 || isValidPropertyFilter(filters[filters.length - 1])) {
+                    return [...filters, {}]
```

Comment:
adding in the empty object is a bit confusing (the name filtersWithNew threw me...)

but it's not new here, just made more visible, so no reason to block

Reply:
Might be nicer to expose `filters.length === 0 || isValidPropertyFilter(filters[filters.length - 1])` and let the UI show the new filter when it is true

Reply:
You mean like exposing `filtersWithNew`? :)

Given the two green checkmarks, I'll merge it in as in.